vulnfeeds/cmd/debian/main.go

```diff
@@ -45,14 +51,97 @@ func main() {
 		Logger.Fatalf("Failed to get Debian distro info data: %s", err)
 	}
 
-	cvePkgInfos := generateDebianSecurityTrackerOSV(debianData, debianReleaseMap)
-	if err = writeToOutput(cvePkgInfos); err != nil {
+	osvCves := generateOSVFromDebianTracker(debianData, debianReleaseMap)
+
+	if err = writeToOutput(osvCves, *debianOutputPath); err != nil {
 		Logger.Fatalf("Failed to write OSV output file: %s", err)
 	}
 
 	Logger.Infof("Debian CVE conversion succeeded.")
 }
 
+// generateOSVFromDebianTracker converts Debian Security Tracker entries to OSV format.
+func generateOSVFromDebianTracker(debianData DebianSecurityTrackerData, debianReleaseMap map[string]string) map[string]*vulns.Vulnerability {
+	Logger.Infof("Converting Debian Security Tracker data to OSV.")
+	osvCves := make(map[string]*vulns.Vulnerability)
+
+	// Sorts packages to ensure results remain consistent between runs.
+	pkgNames := make([]string, 0, len(debianData))
+	for name := range debianData {
+		pkgNames = append(pkgNames, name)
+	}
+	sort.Strings(pkgNames)
+
+	// Sorts releases to ensure pkgInfos remain consistent between runs.
+	releaseNames := make([]string, 0, len(debianReleaseMap))
+	for k := range debianReleaseMap {
+		releaseNames = append(releaseNames, k)
+	}
+
+	sort.Slice(releaseNames, func(i, j int) bool {
+		vi, _ := strconv.ParseFloat(debianReleaseMap[releaseNames[i]], 64)
+		vj, _ := strconv.ParseFloat(debianReleaseMap[releaseNames[j]], 64)
+
+		return vi < vj
+	})
+
+	for _, pkgName := range pkgNames {
+		pkg := debianData[pkgName]
+		for cveID, cveData := range pkg {
+			v, ok := osvCves[cveID]
+			if !ok {
+				v = &vulns.Vulnerability{
+					Vulnerability: osvschema.Vulnerability{
+						ID:        "DEBIAN-" + cveID,
+						Upstream:  []string{cveID},
+						Modified:  time.Now().UTC(),
+						Published: time.Now().UTC(),
```

Comment:
The published time probably shouldn't be when we convert it, but be the CVE publish time.